tests/publiccloud/ssh_interactive_init.pm

```diff
@@ -39,49 +34,44 @@ sub run {
     $args->{my_instance} = $instance;
 
     # configure ssh client, fetch the instance ssh public key, do not use default $instance->ssh_opts
-    assert_script_run('curl ' . data_url('publiccloud/ssh_config') . ' -o ~/.ssh/config');
-    assert_script_run "ssh-keyscan " . $instance->public_ip . " >> ~/.ssh/known_hosts";
+    my $ssh_config = data_url('publiccloud/ssh_config');
+    assert_script_run("curl $ssh_config -o ~/.ssh/config");
+    assert_script_run("ssh-keyscan $instance->{public_ip} >> ~/.ssh/known_hosts");
```

Comment:
It's a `Mojo::Base` class using `has()` so the previous version was more accurate, I would go with the old one.

```suggestion
    assert_script_run('ssh-keyscan ' . $instance->public_ip . ' >> ~/.ssh/known_hosts');
```

Reply:
I don't understand what is the advantage. Do you find the old one more readable?

Reply:
`$instance->public_ip()` is calling a method. And `$instance->{public_ip}` is taking the hash value. So if someone will change the `public_ip()` to call something in advance, it would fail with `hash-style`.
It's cosmetic and about usage of https://docs.mojolicious.org/Mojo/Base#has
It state:
```
Create attribute accessors for hash-based objects
```
So your way would be also ok, if I understand `attr()` implementation of Mojo::Base correct https://docs.mojolicious.org/Mojo/Base.txt

Reply:
Is this ok?
```
assert_script_run(sprintf(q(ssh-keyscan %s >> ~/.ssh/known_hosts), $instance->public_ip));
```

Reply:
for me it looks good. And if you don't like the `q()` syntax here. We could also use the normal ticks

```perl
assert_script_run(sprintf('ssh-keyscan %s >> ~/.ssh/known_hosts', $instance->public_ip));
```